### PR TITLE
[Flysystem 1] Better check if file exists.

### PR DESCRIPTION
**Pull request by Nyholm**
We dont need to check if a directory exists.

### CODE REVIEW

**Comment by jderusse** [comment]
An other optimization: use `objectExists([''])->isSuccess()` or at least `headObject()` to avoid downloading the content (even if the stream is not consumed, I believe some bytes are downloaded and bufferized)

**Comment by Nyholm** [comment]
Nice. I've updated the PR

File: src/Integration/Flysystem/S3/src/S3FilesystemV1.php [hunk 1]
```diff
@@ -173,22 +173,14 @@ public function has($path)
...
 
-        $result = $this->client->getObject(array_merge($this->options, [
+        $result = $this->client->objectExists(array_merge($this->options, [
             'Bucket' => $this->bucket,
             'Key' => $location,
         ]));
 
-        try {
-            $result->resolve();
+        $result->wait();
```
Comment (jderusse) [inline src/Integration/Flysystem/S3/src/S3FilesystemV1.php:181]: `wait` will wait until the object exists.
You should use `isSuccess` or `isFailure` to immediately get if the o ject exists or not.

Comment (Nyholm) [reply src/Integration/Flysystem/S3/src/S3FilesystemV1.php]: Oops, 
Thank you

**Comment by Nyholm** [comment]
Thank you for the reviews.

**Pull request merged**